### PR TITLE
Various bash fixes.

### PR DESCRIPTION
**Pull request by matejak**
Fixed quoting issues, removed IFS manipulation in favor of the readarray bash builtin.

See e.g. https://unix.stackexchange.com/questions/263883/bash-pipe-find-output-into-readarray for example usage of `readarray`.

### CODE REVIEW

File: linux_os/guide/system/software/updating/ensure_oracle_gpgkey_installed/bash/shared.sh [hunk 1]
```diff
@@ -14,7 +14,7 @@ if [ "${RPM_GPG_DIR_PERMS}" -le "755" ]
 then
   # If they are safe, try to obtain fingerprints from the key file
   # (to ensure there won't be e.g. CRC error)
-  IFS=$'\n' GPG_OUT=($(gpg --with-fingerprint --with-colons "$OL_RELEASE_KEY" | grep "^fpr" | cut -d ":" -f 10))
```
Comment (matejak) [inline linux_os/guide/system/software/updating/ensure_oracle_gpgkey_installed/bash/shared.sh]: IFS trick doesn't work at all.

File: shared/bash_remediation_functions/fix_audit_watch_rule.sh [hunk 2]
```diff
@@ -53,40 +54,35 @@ then
...
-		if [ ! -e "$files_to_inspect" ]
+		if [ ! -e "$key_rule_file" ]
 		then
-			touch "$files_to_inspect"
-			chmod 0640 "$files_to_inspect"
+			touch "$key_rule_file"
+			chmod 0640 "$key_rule_file"
 		fi
+
+		files_to_inspect+=("$key_rule_file")
```
Comment (matejak) [inline shared/bash_remediation_functions/fix_audit_watch_rule.sh]: The previous behavior violated comment `# Append '/etc/audit/rules.d/$key.rules' into list of files ...`

File: linux_os/guide/system/permissions/partitions/mount_option_nodev_nonroot_local_partitions/bash/rhel6.sh [hunk 3]
```diff
@@ -65,10 +63,8 @@ do
             # Replace old mount options for particular /etc/fstab's row (for this target
             # and fstype) with new mount options
             sed -i "s#${TARGET_HEAD}\(.*\)${TARGET_TAIL}#${TARGET_HEAD}${MOUNT_OPTIONS}${TARGET_TAIL}#" /etc/fstab
-
+            # Reset IFS back to default
+            IFS="$IFS_BKP"
```
Comment (mildas) [inline linux_os/guide/system/permissions/partitions/mount_option_nodev_nonroot_local_partitions/bash/rhel6.sh]: Why reset `IFS` every iteration?

Comment (matejak) [reply linux_os/guide/system/permissions/partitions/mount_option_nodev_nonroot_local_partitions/bash/rhel6.sh]: Basically this minimizes the time when `IFS` is different from what is the common expectation.

File: linux_os/guide/system/software/integrity/software-integrity/rpm_verification/rpm_verify_ownership/bash/shared.sh [hunk 4]
```diff
@@ -4,29 +4,23 @@
 # complexity = high
 # disruption = medium
 
-# Declare array to hold list of RPM packages we need to correct permissions for
-SETPERMS_RPM_LIST=()
+# Declare array to hold set of RPM packages we need to correct permissions for
+declare -A SETPERMS_RPM_DICT
```
Comment (matejak) [inline linux_os/guide/system/software/integrity/software-integrity/rpm_verification/rpm_verify_ownership/bash/shared.sh]: Changes here are very similar to changes that were done to the tests/data/group_system/group_software/group_integrity/group_software-integrity/group_rpm_verification/rule_rpm_verify_permissions/all_permissions_ok.pass.sh test scenario earier.

File: linux_os/guide/system/software/updating/ensure_fedora_gpgkey_installed/bash/shared.sh [hunk 5]
```diff
@@ -25,13 +25,8 @@ if [ "${RPM_GPG_DIR_PERMS}" -le "755" ]
 then
   # If they are safe, try to obtain fingerprints from the key file
   # (to ensure there won't be e.g. CRC error).
-  # Backup IFS value
-  IFS_BKP="$IFS"
-  IFS=$'\n'
-  GPG_OUT=($(gpg --with-fingerprint --with-colons "${REDHAT_RELEASE_KEY}" | grep '"^fpr' | cut -d ":" -f 10))
+  readarray GPG_OUT < <(gpg --with-fingerprint --with-colons "${REDHAT_RELEASE_KEY}" | grep '"^fpr' | cut -d ":" -f 10)
```
Comment (jan-cerny) [inline linux_os/guide/system/software/updating/ensure_fedora_gpgkey_installed/bash/shared.sh]: I know it was there before, but I have tried this command on my machine and I don't get any line that would start with `"`.  The grep regular expression should be just `^fpr`, without `"` I think. See the similar remediations for RHEL and OL.

Comment (matejak) [reply linux_os/guide/system/software/updating/ensure_fedora_gpgkey_installed/bash/shared.sh]: You are right, other remediations don't have the double quote either.

**Comment by jan-cerny** [comment]
Great! Thanks!

**Pull request merged**